### PR TITLE
Update "node selection" documentation

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
fixes: #2671 
In 1.6 we move node and pod affinity/anti-affinity from pod annotations to the Affinity field of PodSpec, this PR is to update relevant documents.

/cc @kubernetes/sig-scheduling-pr-reviews

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2699)

### CODE REVIEW

File: docs/user-guide/node-selection/index.md [hunk 1]
```diff
@@ -77,16 +77,16 @@ with a standard set of labels. As of Kubernetes v1.4 these labels are
 * `beta.kubernetes.io/os`
 * `beta.kubernetes.io/arch`
 
-## Alpha features: affinity and anti-affinity
+## Beta features: affinity and anti-affinity
```
Comment (davidopp) [inline docs/user-guide/node-selection/index.md]: I would change the title to just "Affinity and anti-affinity". It's fine to mention Beta in the other parts of the doc, but for the title I think simpler is probably better (the Alpha was there as a kind of warning, but now that it's Beta I don't think we need to say anything in the title).

File: docs/user-guide/node-selection/index.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ described in the third item listed above, in addition to having the first and se
 `nodeSelector` continues to work as usual, but will eventually be deprecated, as node affinity can express
 everything that `nodeSelector` can express.
 
-### Node affinity (alpha feature)
+### Node affinity (beta feature)
 
 Node affinity was introduced in Kubernetes 1.2.
```
Comment (davidopp) [inline docs/user-guide/node-selection/index.md]: I would say "introduced as alpha in Kubernetes 1.2"

File: docs/user-guide/node-selection/index.md [hunk 3]
```diff
@@ -116,16 +116,16 @@ Thus an example of `requiredDuringSchedulingIgnoredDuringExecution` would be "on
 and an example `preferredDuringSchedulingIgnoredDuringExecution` would be "try to run this set of pods in availability
 zone XYZ, but if it's not possible, then allow some to run elsewhere".
 
-Node affinity is currently expressed using an annotation on Pod. Once the feature goes to GA it will use a field of Pod.
+Node affinity is specified as field `nodeAffinity` of `affinity` in the PodSpec.
```
Comment (davidopp) [inline docs/user-guide/node-selection/index.md]: s/of/of field/

File: docs/user-guide/node-selection/index.md [hunk 4]
```diff
@@ -157,13 +157,14 @@ in the same zone, since they communicate a lot with each other"
 and an example `preferredDuringSchedulingIgnoredDuringExecution` anti-affinity would be "spread the pods from this service across zones"
 (a hard requirement wouldn't make sense, since you probably have more pods than zones).
 
-Inter-pod affinity and anti-affinity are currently expressed using an annotation on Pod. Once the feature goes to GA it will use a field.
+Inter-pod affinity is specified as field `podAffinity` of `affinity` in the PodSpec.
```
Comment (davidopp) [inline docs/user-guide/node-selection/index.md]: s/of/of field/

File: docs/user-guide/node-selection/index.md [hunk 5]
```diff
@@ -157,13 +157,14 @@ in the same zone, since they communicate a lot with each other"
...
+And inter-pod anti-affinity is specified as field `podAntiAffinity` of `affinity`.
```
Comment (davidopp) [inline docs/user-guide/node-selection/index.md]: s/of/of field/
also, add "in the PodSpec" at the end of the sentence.

**Comment by kevin-wangzefeng** [comment]
@davidopp, refreshed, PTAL :）

**Comment by davidopp** [comment]
LGTM

Thanks!

**Comment by chenopis** [comment]
@kevin-wangzefeng If this is for 1.6, I can move this PR into the `release-1.6` branch and merge it there. Thx

**Comment by davidopp** [comment]
Yes this is for 1.6

**Comment by chenopis** [comment]
Thanks, @davidopp !

**Pull request merged**